### PR TITLE
Make ids of schema objects stable

### PR DESCRIPTION
**Pull request by msullivan**
Instead of generating them randomly, use a uuid5 generated using the
name/type of the object.

Does the implementation work of #6054, but I don't want to close that
(and we shouldn't depend on this) until we have a testing setup.

I tested this using:
```
import pickle
from edb.common import devmode
devmode.enable_dev_mode()

p1 = pickle.loads(open('backend-stdlib.pickle', mode='rb').read()[20:])
p2 = pickle.loads(open('build/cache/backend-stdlib.pickle', mode='rb').read()[20:])

mismatched = [k for k in p1.reflschema._id_to_data if p1.reflschema._id_to_data.get(k) != p2.reflschema._id_to_data.get(k)]

print("reflschema:", p1.reflschema.__dict__ == p2.reflschema.__dict__)
print("stdschema:", p1.stdschema.__dict__ == p2.stdschema.__dict__)
```

### CODE REVIEW

File: edb/schema/objects.py [hunk 1]
```diff
@@ -1101,7 +1104,15 @@ def _prepare_id(
...
-            return cls.generate_id(schema, data)
+            if stdmode:
+                # When compiling the standard library, we generate
+                # stable ids based on the internal name and the type's
+                # name. This keeps std schemas compatible across
+                # minor versions at least.
+                return uuidgen.uuid5(
+                    TYPE_ID_NAMESPACE, f'{name}-{cls.__name__}')
+            else:
+                return cls.generate_id(schema, data)
```
Comment (elprans) [inline edb/schema/objects.py:1115]: We don't seem to actually overload `generate_id` anywhere anymore, do you want to get rid of it while you're here?

Comment (elprans) [reply edb/schema/objects.py:1115]: Or, if you want to keep it for some reason, I'd push the `stdmode` thing into it.

Comment (msullivan) [reply edb/schema/objects.py]: Yeah we'll just drop it

**Review by elprans — APPROVED** [review]
Needs a catver bump

**Pull request merged**